src/Middleware/src/SitecoreExtensions/code/Extensions/SitecoreLoggingExtensions.cs

```diff
@@ -0,0 +1,343 @@
+﻿using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Collections.Generic;
+using Sitecore.Foundation.SitecoreExtensions.Extensions;
+
+namespace Sitecore.Diagnostics
+{
+	public static class LogExt
+	{
+		/// <summary>
+		/// Common re-usable LogException method used for logging exception separated log errors throughout the entire application into Sitecore Log files
+		/// </summary>
+		/// <param name="appLogFileKey"></param>
+		/// <param name="methodName"></param>
+		/// <param name="message"></param>
+		/// <param name="tryCatchMessage"></param>
+		/// <param name="errorTraceCert"></param>
+		/// <param name="obj"></param>
+		/// <param name="createCustomLogFile"></param>
+		public static void LogException(string appLogFileKey, string methodName, string message, string tryCatchMessage, string errorTraceCert, object obj, bool createCustomLogFile = false)
+		{
+			message = (message.Contains("[methodName]")) ? message.Replace("[methodName]", methodName) : message;
+			if (createCustomLogFile)
+			{
+				LoggingNotifications.LogExceptionNotification(appLogFileKey, methodName, message, tryCatchMessage, errorTraceCert);
+			}
+			else
+			{
+				var exception = LoggingNotifications.GetLogExceptionMessage(methodName, message, tryCatchMessage, errorTraceCert, LoggingNotifications.GetApiResponseMessagePrefixKey());
+				Log.Error($@"{message} - {LoggingNotifications.GetExceptionMessagePrefixKey()}: {exception}.", obj);
+				throw new Exception(exception);
+			}            
+		}
+
+		/// <summary>
+		/// Common re-usable LogApiResponseMessages method used for api response message logging throughout the entire application
+		/// </summary>
+		/// <param name="methodName"></param>
+		/// <param name="message"></param>
+		/// <param name="messageKeyValue"></param>
+		/// <param name="isError"></param>
+		public static void LogApiResponseMessages(string methodName, string message, string messageKeyValue, bool isError = false)
+		{
+			if (string.IsNullOrEmpty(message))
+			{
+				return;
+			}
+
+			messageKeyValue = (!string.IsNullOrEmpty(messageKeyValue)) ? messageKeyValue : LoggingNotifications.GetGeneralLogMessagePrefixKey();
+			message = LoggingNotifications.GetLogExceptionMessage(methodName, $@"{messageKeyValue}: {message}", string.Empty, string.Empty, LoggingNotifications.GetApiResponseMessagePrefixKey());
+			if (isError)
+			{
+				Log.Error(message, new object());
+				throw new Exception(message);
```

Comment:
Can you please rethrow the original exception instead of throwing a new exception? The former retains the original stacktrace, while this one does not. Should be simply `throw;` instead of `throw new Exception(message);`;

Details here: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2200

Reply:
Resolved this request.